src/components/core/menu/mainMenu/MainMenuRightChainSelector.tsx

```diff
@@ -0,0 +1,74 @@
+import { ReactComponent as IconCheck } from 'assets/icons/check.svg';
+import { FC, useState } from 'react';
+import { DropdownMenu } from 'components/common/dropdownMenu';
+import { buttonStyles } from 'components/common/button/buttonStyles';
+import { carbonEvents } from 'services/events';
+import { cn } from 'utils/helpers';
+
+interface Props {
+  networks: {
+    id: string;
+    name: string;
+    logoUrl: string;
+    isCurrentNetwork: boolean;
+    selectNetwork: () => void;
+  }[];
+}
+
+export const MainMenuRightChainSelector: FC<Props> = ({ networks }) => {
+  const [isOpen, setIsOpen] = useState(false);
```

Comment:
I think that you don't need to pass this state if you only use it within the dropdown menu & trigger. To verfiy